examples/appSpec/zomes/three/capabilities/main/code/index.ts

```diff
@@ -38,6 +42,54 @@ export function test_get(encoded_allocation: u32): u32 {
 }
 
 
+/*----------  test_decode_params boilerplate  ----------*/
+
+
+var valueString: string = "";
+
+class ParameterHandler extends Handler {
+  onString(keyStack: Array<string>, value: string): void {
+    if(keyStack[keyStack.length-1] == "keyString") {
+        valueString = value;
```

Comment:
All JSON callbacks now receive a keyStack array as the first parameter. The last element is the current key and the previous keys give the nested objects it is currently in

Reply:
nice 👍